libraries/core_libs/network/src/tarcap/packets_handlers/get_pbft_sync_packet_handler.cpp

```diff
@@ -79,10 +79,22 @@ void GetPbftSyncPacketHandler::sendPbftBlocks(dev::p2p::NodeID const &peer_id, s
     }
 
     dev::RLPStream s;
-    s.appendList(3);
-    s << (pbft_chain_synced && last_block);
-    s << last_block;
-    s.appendRaw(data);
+    if (pbft_chain_synced && last_block) {
+      s.appendList(4);
+      s << (pbft_chain_synced && last_block);
+      s << last_block;
+      s.appendRaw(data);
+      const auto votes = db_->getLastBlockCertVotes();
+      s.appendList(votes.size());
+      for (const auto &vote : votes) {
+        s.appendRaw(vote->rlp(true));
+      }
+    } else {
+      s.appendList(3);
+      s << (pbft_chain_synced && last_block);
+      s << last_block;
+      s.appendRaw(data);
+    }
```

Comment:
I think that's dangerous here.

For example:
1. N1 has PBFT chain size 2
2. N2 sync from N1
3. N1 send B1 to N2, then lose connection to N2.
4. N2 will find another peer to sync
5. In the meanwhile, N2 will process B1 from syncing queue
6. N2 failed processing B1 because no cert votes for B1
7. N2 will not receive B1 again, because N2 has marked B1 as known block

Reply:
B1 will stay in the queue until we get B2, it will not be removed from the queue